nixos/modules/system/boot/luksroot.nix

```diff
@@ -695,6 +697,12 @@ in
                   description = "Length of the LUKS slot key derived with PBKDF2 in byte.";
                 };
 
+                renewKey = mkOption {
+                  default = true;
+                  type = types.bool;
+                  description = "Whether to regenerate a new LUKS key in the header (true), or do not regenerate it (false).";
+                };
```

Comment:
Code wise looks ok to me, but is there any security implications for setting this false? If yes, I think it would be better to have some warning here.

Reply:
Not really, it just makes the LUKS header persisting but it is already almost impossible to brute force it, so no need to renew it every time.
It is pretty useful if you want to backup it in case you forget the password.
Some other LUKS setups for instance on Ubuntu are not renewing the header by default.